spec.bs

```diff
@@ -116,6 +132,49 @@ if (document.wasDiscarded) {
 }
 </pre>
 
+Feature Policies {#feature-policies}
+=====================
+To improve CPU utilization of [=nested browsing contexts=] that are not critical to a page's usability two feature policies are defined:
+ * execution-while-not-rendered, which has a [=default allowlist=] of '*'
+ * execution-while-out-of-viewport, which has a [=default allowlist=] of '*'
+
+The <dfn export>execution-while-not-rendered</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for [=nested browsing contexts=] whose [=browsing context container=] is not [=being rendered=]. ie. `display: none`.
+
+The <dfn export>execution-while-out-of-viewport</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for [=nested browsing contexts=] whose [=browsing context container=] does not intersect the [=viewport=] according
+to [=compute the intersection of a target element and the root=].
+
+Application of these policies is accomplished by placing the document (and its decendants) in a [=/frozen=] state when the following
+conditions have been met:
+ * [=iframe load event steps=] have been run
```

Comment:
What if the load events are currently being run? Will this process wait until loading is complete and apply the frozen state after?

Reply:
Yes. There is a monkey patch to the iframe load steps that apply update document frozeness. Is that sufficient for you?

Reply:
Not sure I fully understand "monkey patching" in this context, but I expect it would work as long as we control for races. E.g. autoplay should not trigger before frozen state is applied.

Reply:
Yes without the posted task source it should be good.